x-pack/plugins/search_playground/public/components/search_mode/result_list.tsx

```diff
@@ -5,54 +5,114 @@
-  { id: '123321', name: 'John Doe', age: 25 },
-  { id: '123321', name: 'John Doe', age: 25 },
-  { id: '123321', name: 'John Doe', age: 25 },
-  { id: '123321', name: 'John Doe', age: 25 },
-  { id: '123321', name: 'John Doe', age: 25 },
-  { id: '123321', name: 'John Doe', age: 25 },
-  { id: '123321', name: 'John Doe', age: 25 },
-  { id: '123321', name: 'John Doe', age: 25 },
-];
+import { UnifiedDocViewerFlyout } from '@kbn/unified-doc-viewer-plugin/public';
+
+import type { DataView } from '@kbn/data-views-plugin/public';
+import type { EsHitRecord } from '@kbn/discover-utils/types';
+import type { SearchHit } from '@elastic/elasticsearch/lib/api/types';
+import { buildDataTableRecord } from '@kbn/discover-utils';
+import { i18n } from '@kbn/i18n';
+import { Pagination } from '../../types';
+import { getPageCounts } from '../../utils/pagination_helper';
+import { EmptyResults } from './empty_results';
+import { useKibana } from '../../hooks/use_kibana';
+
+export interface ResultListArgs {
+  searchResults: SearchHit[];
+  pagination: Pagination;
+  onPaginationChange: (nextPage: number) => void;
+}
 
-export const ResultList: React.FC = () => {
+export const ResultList: React.FC<ResultListArgs> = ({
+  searchResults,
+  pagination,
+  onPaginationChange,
+}) => {
+  const {
+    services: { data },
+  } = useKibana();
+  const [dataView, setDataView] = useState<DataView | null>(null);
+  useEffect(() => {
+    data.dataViews.getDefaultDataView().then((d) => setDataView(d));
+  }, [data]);
+  const [flyoutDocId, setFlyoutDocId] = useState<string | undefined>(undefined);
+  const { totalPage, page } = getPageCounts(pagination);
+  const hit =
+    flyoutDocId &&
+    buildDataTableRecord(searchResults.find((item) => item._id === flyoutDocId) as EsHitRecord);
   return (
     <EuiPanel grow={false}>
       <EuiFlexGroup direction="column" gutterSize="none">
-        {DEMO_DATA.map((item, index) => {
+        {searchResults.map((item, index) => {
           return (
             <>
-              <EuiFlexItem key={item.id + '-' + index} grow>
+              <EuiFlexItem
+                key={item._id + '-' + index}
+                onClick={() => setFlyoutDocId(item._id)}
+                grow
+              >
                 <EuiFlexGroup direction="column" gutterSize="xs">
                   <EuiFlexItem grow>
                     <EuiTitle size="xs">
-                      <h2>{item.id}</h2>
+                      <h3>ID:{item._id}</h3>
                     </EuiTitle>
                   </EuiFlexItem>
                   <EuiFlexItem grow>
                     <EuiText size="s">
-                      <p>{item.name}</p>
+                      <p>
+                        {i18n.translate('xpack.searchPlayground.resultList.result.score', {
+                          defaultMessage: 'Document score: {score}',
+                          values: { score: item._score },
+                        })}
+                      </p>
```

Comment:
@joemcelroy, Will it be possible to display `score` with `kbn-search-index-documents`? Maybe as `fieldValue`?

Reply:
Not current AFAIK but im hoping we will extend `kbn-search-index-documents` component to accommodate this + a number of other features.